presto-hive/src/test/java/com/facebook/presto/hive/parquet/write/TestDataWritableWriter.java

```diff
@@ -0,0 +1,378 @@
+            case DOUBLE:
+                recordConsumer.addDouble(((DoubleObjectInspector) inspector).get(value));
+                break;
+            case BOOLEAN:
+                recordConsumer.addBoolean(((BooleanObjectInspector) inspector).get(value));
+                break;
+            case FLOAT:
+                recordConsumer.addFloat(((FloatObjectInspector) inspector).get(value));
+                break;
+            case BYTE:
+                recordConsumer.addInteger(((ByteObjectInspector) inspector).get(value));
+                break;
+            case INT:
+                recordConsumer.addInteger(((IntObjectInspector) inspector).get(value));
+                break;
+            case LONG:
+                recordConsumer.addLong(((LongObjectInspector) inspector).get(value));
+                break;
+            case SHORT:
+                recordConsumer.addInteger(((ShortObjectInspector) inspector).get(value));
+                break;
+            case STRING:
+                String v = ((StringObjectInspector) inspector).getPrimitiveJavaObject(value);
+                recordConsumer.addBinary(Binary.fromString(v));
+                break;
+            case CHAR:
+                String vChar = ((HiveCharObjectInspector) inspector).getPrimitiveJavaObject(value).getStrippedValue();
+                recordConsumer.addBinary(Binary.fromString(vChar));
+                break;
+            case VARCHAR:
+                String vVarchar = ((HiveVarcharObjectInspector) inspector).getPrimitiveJavaObject(value).getValue();
+                recordConsumer.addBinary(Binary.fromString(vVarchar));
+                break;
+            case BINARY:
+                byte[] vBinary = ((BinaryObjectInspector) inspector).getPrimitiveJavaObject(value);
+                recordConsumer.addBinary(Binary.fromByteArray(vBinary));
+                break;
+            case TIMESTAMP:
+                Timestamp ts = ((TimestampObjectInspector) inspector).getPrimitiveJavaObject(value);
+                recordConsumer.addBinary(NanoTimeUtils.getNanoTime(ts, false).toBinary());
+                break;
+            case DECIMAL:
+                HiveDecimal vDecimal = ((HiveDecimal) inspector.getPrimitiveJavaObject(value));
+                DecimalTypeInfo decTypeInfo = (DecimalTypeInfo) inspector.getTypeInfo();
+                recordConsumer.addBinary(decimalToBinary(vDecimal, decTypeInfo));
+                break;
+            case DATE:
+                Date vDate = ((DateObjectInspector) inspector).getPrimitiveJavaObject(value);
+                recordConsumer.addInteger(DateWritable.dateToDays(vDate));
+                break;
+            default:
+                throw new IllegalArgumentException("Unsupported primitive data type: " + inspector.getPrimitiveCategory());
+        }
+    }
+
+    private Binary decimalToBinary(final HiveDecimal hiveDecimal, final DecimalTypeInfo decimalTypeInfo)
+    {
+        int prec = decimalTypeInfo.precision();
```

Comment:
`prec` -> `precision`

Reply:
the same as above, need your opinion regarding updating copy of Hive Parquet writer